lua/telescope/_extensions/frecency.lua

```diff
@@ -1,298 +1,37 @@
-        directory_col_width = #(utils.path_tail(state.active_filter)) + 1
-      else
-        directory_col_width = #(Path:new(state.active_filter):make_relative(os_home)) + 1
-      end
-    end
-    local res = {}
-    res[1] = state.show_scores and { width = 8 } or nil
-    if state.show_filter_column then
-      table.insert(res, { width = directory_col_width })
-    end
-    if has_devicons and not state.disable_devicons then
-      table.insert(res, { width = 2 }) -- icon column
-    end
-    table.insert(res, { remaining = true })
-    return res
-  end
-
-  local displayer = entry_display.create {
-    separator = "",
-    hl_chars = { [os_path_sep] = "TelescopePathSeparator" },
-    items = get_display_cols(),
-  }
-
-  local bufnr, buf_is_loaded, display_filename, hl_filename, display_items, icon, icon_highlight
-  local make_display = function(entry)
-    bufnr = vim.fn.bufnr
-    buf_is_loaded = vim.api.nvim_buf_is_loaded
-    display_filename = entry.name
-    hl_filename = buf_is_loaded(bufnr(display_filename)) and "TelescopeBufferLoaded" or ""
-    display_filename = format_filepath(display_filename, opts)
-
-    display_items = state.show_scores and { { entry.score, "TelescopeFrecencyScores" } } or {}
-
-    -- TODO: store the column lengths here, rather than recalculating in get_display_cols()
-    -- TODO: only include filter_paths column if opts.show_filter_col is true
-    local filter_path = ""
-    if state.active_filter then
-      if state.active_filter_tag == "LSP" or state.active_filter_tag == "CWD" then
-        filter_path = utils.path_tail(state.active_filter) .. os_path_sep
-      else
-        filter_path = Path:new(state.active_filter):make_relative(os_home) .. os_path_sep
-      end
-    end
-    table.insert(display_items, { filter_path, "Directory" })
-    if has_devicons and not state.disable_devicons then
-      icon, icon_highlight = devicons.get_icon(entry.name, string.match(entry.name, "%a+$"), { default = true })
-      table.insert(display_items, { icon, icon_highlight })
-    end
-    table.insert(display_items, { display_filename, hl_filename })
-
-    return displayer(display_items)
-  end
-
-  local update_results = function(filter)
-    local filter_updated = false
-
-    -- validate tag
-    local ws_dir = filter and state.user_workspaces[filter]
-    if filter == "LSP" and not vim.tbl_isempty(state.lsp_workspaces) then
-      ws_dir = state.lsp_workspaces[1]
-    end
-
-    if filter == "CWD" then
-      ws_dir = state.cwd
-    end
-
-    if ws_dir ~= state.active_filter then
-      filter_updated = true
-      state.active_filter = ws_dir
-      state.active_filter_tag = filter
-    end
-
-    if vim.tbl_isempty(state.results) or filter_updated then
-      state.results = db_client.get_file_scores(state.show_unindexed, ws_dir)
+p.exports = {
+  frecency = picker.fd,
+  get_workspace_tags = picker.workspace_tags, --TODO: what is the use case for this?
```

Comment:
I think this shouldn't be public. It returns a list of defined tags. You remember doing `:<tab>` in prompt and then you can prefilter for a workspace? I think you can remove both function and this like